[TASK] Describe basic Extbase URI argument naming (#4525)
TYPO3-Documentation/TYPO3CMS-Reference-CoreApi: Documentation/ExtensionArchitecture/Extbase/Reference/UriArguments.rst

> .. include:: /Includes.rst.txt

Reviewer: 4 spaces indent

Author: Thanks! Copied another file which still had the old indentation.

.. index:: Extbase; URI arguments; arguments

===================================
URI arguments and reserved keywords
===================================

Extbase uses special URI arguments to pass variables
to Controller arguments and the framework itself.

.. todo: Check for completion of this

Extbase uses a prefixed URI argument scheme that relies
on plugin configuration.

For example, the example extension `EXT:blog_example` would use:

..  code-block:: plaintext

>     https://example.com/blog/?tx_blogexample_bloglist[action]=show&tx_blogexample_bloglist[controller]=Post&tx_blogexample_bloglist[post]=4711&cHash=...

Reviewer: IN some of our documentation I saw that we should use `example.org` at first

>     https://example.org/blog/?tx_blogexample_bloglist[action]=show&tx_blogexample_bloglist[controller]=Post&tx_blogexample_bloglist[post]=4711&cHash=...

Reviewer 2: Can we break this into several lines? This causes horizontal scrolling

Author: Yes, good idea!

    // Actually, the [] parameters are often URI encoded, so this is emitted:
    https://example.com/blog/?tx_blogexample_bloglist%5Baction%5D=show&tx_blogexample_bloglist%5Bcontroller%5D=Post&tx_blogexample_bloglist%5Bpost%5D=4711&cHash=...

as the created URI to execute the `showAction` of the Controller `PostController`
within the plugin `BlogList`.

The following arguments are evaluated:

*   `tx_(extensionName)_(pluginName)[action]`: Controller action to execute
*   `tx_(extensionName)_(pluginName)[controller]`: Controller containing the action

> *   `tx_(extensionName)_(pluginName)[format]`: Output format (usually HTML, can also be JSON or custom types)

Reviewer: Format should be lowercase. So `html` or `json`

Author: Ack, thanks!

> *   `cHash` - the cHash is always calculated to validate the URI is allowed to be called (see :ref:`chash`)

Reviewer: ```suggestion
*   `cHash` - the cHash is always calculated to validate that the URI can be called. (see :ref:`chash`)
```

Author: I think it is important to differentiate here that chash doesn't prevent a URI from being called, it just results in a "denied access" check. So I would like to keep the "allowed to" wording here, would that be ok with you?

Reviewer: I only had a problem with the two `is` in the original sentence.
I don't have any problem to keep `is allowed to`

Author: Ah, I understand. I committed a slightly modified version now, how about that? :)

> *   `tx_(extensionName)_(pluginName)[action]`: Controller action to execute
> *   `tx_(extensionName)_(pluginName)[controller]`: Controller containing the action
> *   `tx_(extensionName)_(pluginName)[format]`: Output format (usually `html`, can also be `json` or custom types)
> *   `cHash` - the cHash always gets calculated to validate that the URI is allowed to be called. (see :ref:`chash`)

Reviewer 2: ```suggestion
 `tx_(extensionName)_(pluginName)[action]`: 
     Controller action to execute
`tx_(extensionName)_(pluginName)[controller]`
    Controller containing the action
`tx_(extensionName)_(pluginName)[format]`
    Output format (usually `html`, can also be `json` or custom types)
`cHash`
    the cHash always gets calculated to validate that the URI is allowed to be called. (see :ref:`chash`)
```
I would suggest to use a definition list for easier readability and better speech support

Any other argument will be passed along to the controller action and can be
retrieved via :php:`$this->request->getArgument()`. Usually this is auto-wired
by the automatic parameter mapping of Extbase.

These URI arguments can also be used for the routing configuration, see
:ref:`routing-extbase-plugin-enhancer`.

..  warning::

    The listed keys `action`, `controller` and `format` are reserved keywords.
    Never use these as custom argument names to your controller actions,
    so instead of :fluid:`<f:uri.action action="new" arguments="{format: someVariable}">`
    you should use :fluid:`<f:uri.action action="new" arguments="{customFormat: someVariable}">`.
    Else, using an argument like this would lead to TYPO3 exceptions of unresolvable
    Fluid template files!

When submitting a HTML :html:`<form>`, the same URI arguments will be part of
a HTTP POST request, with some more special ones:

> *   `tx_(extensionName)_(pluginName)[__referrer]`: An array with information
>     of the referring call (subkeys: `@extension`, `@controller`, `@action`,
>     `arguments` (hashed), `@request` (json)
> *   `tx_(extensionName)_(pluginName)[__trustedProperties]`: List of properties to be submitted to an action (hashed and secured)

Reviewer 2: ```suggestion
 `tx_(extensionName)_(pluginName)[__referrer]`
     An array with information
     of the referring call (subkeys: `@extension`, `@controller`, `@action`,
    `arguments` (hashed), `@request` (json)
`tx_(extensionName)_(pluginName)[__trustedProperties]`
    List of properties to be submitted to an action (hashed and secured)
```
